Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraPreview/CameraPreviewPage.xaml.cs

```diff
@@ -0,0 +1,159 @@
+        private SoftwareBitmapSource _softwareBitmapSource;
+        private CameraPreview _cameraPreviewControl;
+        private Image _imageControl;
+        private TextBlock _errorMessageText;
+
+        public CameraPreviewPage()
+        {
+            this.InitializeComponent();
+            semaphoreSlim = new SemaphoreSlim(1);
+        }
+
+        public async void OnXamlRendered(FrameworkElement control)
+        {
+            // Using a semaphore lock for synchronocity.
+            // This method gets called multiple times when accessing the page from Latest Pages
+            // and creates unused duplicate references to Camera and memory leaks.
+            await semaphoreSlim.WaitAsync();
+
+            var cameraHelper = _cameraPreviewControl?.CameraHelper;
+            UnsubscribeFromEvents();
+
+            _cameraPreviewControl = control.FindChild<CameraPreview>();
+            if (_cameraPreviewControl != null)
+            {
+                _cameraPreviewControl.PreviewFailed += CameraPreviewControl_PreviewFailed;
+                await _cameraPreviewControl.StartAsync(cameraHelper);
+                _cameraPreviewControl.CameraHelper.FrameArrived += CameraPreviewControl_FrameArrived;
+            }
+
+            _imageControl = control.FindDescendantByName("CurrentFrameImage") as Image;
+            if (_imageControl != null)
+            {
+                _softwareBitmapSource = new SoftwareBitmapSource();
+                _imageControl.Source = _softwareBitmapSource;
+            }
+
+            _errorMessageText = control.FindDescendantByName("ErrorMessage") as TextBlock;
+
+            semaphoreSlim.Release();
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            Shell.Current.RegisterNewCommand("Capture Current Frame", CaptureButton_Click);
+            Application.Current.Suspending += Application_Suspending;
+            Application.Current.Resuming += Application_Resuming;
+        }
+
+        protected async override void OnNavigatedFrom(NavigationEventArgs e)
+        {
+            Application.Current.Suspending -= Application_Suspending;
+            Application.Current.Resuming -= Application_Resuming;
+            await CleanUpAsync();
+        }
+
+        private async void Application_Suspending(object sender, SuspendingEventArgs e)
+        {
+            if (Frame.CurrentSourcePageType == typeof(CameraPreviewPage))
+            {
+                var deferral = e.SuspendingOperation.GetDeferral();
+                await CleanUpAsync();
+                deferral.Complete();
+            }
+        }
+
+        private async void Application_Resuming(object sender, object e)
+        {
+            var cameraHelper = _cameraPreviewControl?.CameraHelper;
+            _cameraPreviewControl.PreviewFailed += CameraPreviewControl_PreviewFailed;
+            await _cameraPreviewControl.StartAsync(cameraHelper);
```

Comment:
Camera Helper is exposed so that the user can access properties and events exposed directly from the helper. The developer can either provide a camera helper if they have one created with some settings or use StartAsync() which creates a default camera helper instance.